diff --git a/src/fatlabel.c b/src/fatlabel.c
index 8397a97..d45fb95 100644
--- a/src/fatlabel.c
+++ b/src/fatlabel.c
@@ -119,8 +119,8 @@ int main(int argc, char *argv[])
 	    /* don't know if here should be more strict !uppercase(label[i]) */
 	    if (islower(label[i])) {
 		fprintf(stderr,
-			"fatlabel: labels cannot contain lower case characters\n");
-		exit(1);
+			"fatlabel: warning - lowercase labels might not work properly with DOS or Windows\n");
+		break;
 	    }
 	rw = 1;
     }
diff --git a/src/mkfs.fat.c b/src/mkfs.fat.c
index cfa3288..9e960ba 100644
--- a/src/mkfs.fat.c
+++ b/src/mkfs.fat.c
@@ -1536,8 +1536,13 @@ int main(int argc, char **argv)
 
 	case 'n':		/* n : Volume name */
 	    sprintf(volume_name, "%-11.11s", optarg);
-	    for (i = 0; i < 11; i++)
-		volume_name[i] = toupper(volume_name[i]);
+	    for (i = 0; volume_name[i] && i < 11; i++)
+		/* don't know if here should be more strict !uppercase(label[i]) */
+		if (islower(volume_name[i])) {
+		    fprintf(stderr,
+		            "mkfs.fat: warning - lowercase labels might not work properly with DOS or Windows\n");
+		    break;
+		}
 
 	    break;
 
