--- dosfstools-3.0.9/src/boot.c.old	2010-01-23 10:13:11.000000000 +0100
+++ dosfstools-3.0.9/src/boot.c	2010-10-06 13:50:55.028885433 +0200
@@ -59,7 +59,7 @@
     { 0xff, "5.25\" 320k floppy 2s/40tr/8sec" },
 };
 
-#if defined __alpha || defined __arm || defined __ia64__ || defined __s390x__ \
+#if defined __alpha || defined __arm || defined __ia64__ \
  || defined __x86_64__ || defined __ppc64__ || defined __bfin__ \
  || defined __MICROBLAZE__
 /* Unaligned fields must first be copied byte-wise */
